016-kafka-connect-build.md

```diff
@@ -0,0 +1,137 @@
+# Kafka Connect Build
+
+This proposal improves our Kafka Connect deployment with support for _build_ for adding additional connector plugins to the Connect deployment.
+
+## Current situation
+
+Currently, Strimzi has two Kafka Connect deployments: `KafkaConnect` and `KafkaConnectS2I`. 
+
+`KafkaConnect` currently requires that the user to manually prepare their own container image with any additional connector plugins they want to use.
+They have to write their own `Dockerfile` which uses the Strimzi image as a base image and adds the additional connectors.
+Then they have to build it and push into some container registry and configure the Kafka Connect deployment to use it.
+
+The `KafkaConnectS2I` is supported only on OKD / OpenShift and uses the S2I build to build the new container image with additional connector plugins.
+It is not support on Kubernetes since pure Kubernetes do not have any support for S2I builds.
+It makes it a bit easier to prepare the new image, but only slightly.
+User has to prepare a local directory with the desired connector plugins and pass it to the S2I build.
+The S2I build adds it to the image and uses it automatically in the Connect deployment.
+In order to add more connector plugins later, user needs to have the old connectors as well as the new connectors - it is not possible to just add the new connectors.
+
+Unlike other Strimzi components, none of these works in a declarative way.
+
+## Motivation
+
+One of the aims of Strimzi is to make using Kafka on Kubernetes as native as possible.
+A big part of that is being able to configure things in a declarative way.
+That is currently missing for adding connector plugins to the Connect deployments.
+
+Being able to configure the connector plugins in the `KafkaConnect` CR will make it easier to use for our users since they will not need to build the container images manually.
+It will also make it easier to to have connector catalog on the website since we will be able to just share the KafkaConnect CR including the build section to add the connector.
+
+## Proposal
+
+A new section named `build` would be added to the `KafkaConnect` custom resource.
+This section will allow users to configure:
+* connector plugins
+* build output
+
+The connector plugins section will be a list where one or more connectors can be defined.
+Each connector consists of a name and list of artifacts which should be downloaded.
+The artifacts can be of different types.
+For example:
+* `jar` for a directly download of Java JARs. 
+* `tgz` or `zip` to _download and unpack_ the artifacts. 
```

Comment:
I guess that perhaps tar.gz could include non-regular files/directories, such as symlinks. I guess we'd need a way to detect those and deal with them, since they shouldn't be allowed.

Reply:
Updated.